### ISSUE TITLE
domino2

### ISSUE THREAD

**Issue by jmitchell81** [Bioconductor/Contributions#3259]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/FertigLab/domino2

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by lshep** [comment]
I'm slightly concerned with the name.  The currently active [domino package on CRAN](https://cran.r-project.org/web/packages/domino/index.html) seems to have completely different functionality?

We also like to have a clean branch to add into the Bioconductor git repository.  There are additional files and directories like pkgdown,  old.README, scenic_bash, .github/workflows.  We understand these tools can be useful and recommend moving them to a different branch on github.  Can the directory structure be cleaned up?

You seem to download a lot of data in the vignette. Is this all example data?  it will likely be worth while to cache the data files being download especially since the Bioconductor builders have build and check time constraints.  See [BiocFileCache](https://bioconductor.org/packages/release/bioc/html/BiocFileCache.html) 

All vignettes should be rendered through the R cmd build of a package and products should not be included in the docs directory. 

Currently when I try to do R CMD build locally it fails 

```
orikern@jbcj433:~/PkgReview$ R CMD build domino2 
* checking for file 'domino2/DESCRIPTION' ... OK
* preparing 'domino2':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
--- re-building ‘domino2.Rmd’ using rmarkdown
trying URL 'https://zenodo.org/records/10222767/files/pbmc_seurat.rds'
Content type 'application/octet-stream' length 294035980 bytes (280.4 MB)
===========================================
downloaded 241.4 MB


Quitting from lines 35-80 [files] (domino2.Rmd)
Error: processing vignette 'domino2.Rmd' failed with diagnostics:
download from 'https://zenodo.org/records/10222767/files/pbmc_seurat.rds' failed
--- failed re-building ‘domino2.Rmd’

--- re-building ‘domino_object_vignette.Rmd’ using rmarkdown
trying URL 'https://zenodo.org/records/10222767/files/pbmc_domino_built.rds'
Content type 'application/octet-stream' length 287884913 bytes (274.5 MB)
==================================================
downloaded 274.5 MB

--- finished re-building ‘domino_object_vignette.Rmd’

--- re-building ‘plotting_vignette.Rmd’ using rmarkdown
trying URL 'https://zenodo.org/records/10222767/files/pbmc_domino_built.rds'
Content type 'application/octet-stream' length 287884913 bytes (274.5 MB)
==================================================
downloaded 274.5 MB

Warning in doTryCatch(return(expr), name, parentenv, handler) :
  semi-transparency is not supported on this device: reported only once per page
Warning in doTryCatch(return(expr), name, parentenv, handler) :
  semi-transparency is not supported on this device: reported only once per page
--- finished re-building ‘plotting_vignette.Rmd’

SUMMARY: processing the following file failed:
  ‘domino2.Rmd’

Error: Vignette re-building failed.
Execution halted

```

**Comment by lshep** [comment]
@jmitchell81 do you have comments or updates to the above?

**Comment by jmitchell81** [comment]
Thank you @lshep, we have been working to address your comments. In regards to the naming of the package, we have suggested alternative names to the investigators that originally wrote and published this package to ensure continuity with the published name of "domino" that has been used in subsequent publications while distinguishing from the package on CRAN and the "DominoEffect" package on Bioconductor. 

We have been able to implement the BiocFileCache suggestion you provided for our vignettes to ensure the data for the vignette can be accessed without timing out. This timeout is likely the reason you were unable to render the vignette as we have been able to successfully build the package with GitHub actions in it's current repository in the FertigLab organization.

For the additional docs in the package, we are working to maintain documentation and interactive vignettes with pkgdown on a separate branch and maintain a dedicated clean branch where pushes to Bioconductor could be made. For this submission of our package, can we specify a branch of our GitHub repository for the submission rather that the HEAD branch of the repository?

Thank you for your help.

**Comment by lshep** [comment]
Please keep us informed about the naming.  Its difficult to rename a package after it is accepted to Bioconductor.  
I will check the updates on BiocFileCache and try to rebuild/recheck later today.   
As far as the branches, our automated scripts that we use to place a package on our system uses the defined "default" branch of the github.  If possible to change the clean branch as default for a day or two when we are agreed to move the package forward in the review process; after that point you can change back to whatever default branch you would like on github as Bioconductor will only build off code pushed to git.bioconductor.org. Hopefully this would be agreeable?

**Comment by jmitchell81** [comment]
Thank you @lshep. We have prepared a branch for the package called "bioconductor" that includes the caching updates and removed the additional documentation from pkgdown. The name is still set as "domino2" for the moment as I am still debugging a unit test that compared results to results from an older version of the package. We still intend upon updating the package's name, but would appreciate your review of the changes made to ensure the package will build for you. The default branch will be set to "bioconductor" for your review for the rest of today.

**Comment by lshep** [comment]
**General**

- [ ] It is a lot of work to clean up and rename packages. We will not ingest
  this into our system until the naming is decided upon to avoid unnecessary
  work for multiple people on our team in the future. 

That being said, I did glance at the package and have the following additional
comments: 

**vignette**

- [ ] Please show things like options and set.seeds to users so they can
  accurately run and reproduce the code in the vignette.

- [ ] We don't allow data to be on github. can the data retrieved from github
  also be moved to a trusted site like zenodo.

- [ ] Vignette code must be executable. Currently no code is evaluated. 


**General**

- [ ] There is currently no interop with Bioconductor standard classes for
  single cell.  Please adjust package to also account for this and show
  examples. See SingleCellExperiment packages/class.

- [ ] Packages used in vignettes and examples must be on CRAN or Bioconductor.

**Comment by jmitchell81** [comment]
Thank you for your feedback @lshep. I have a question for clarification about the data on GitHub. In our current vignettes and in some unit tests, we do retrieve data from Zenodo that is evaluated and processed through this vignette. The code that is evaluated in the R markdown file for our vignette does not use any data hosted on GitHub. We do include code that is not evaluated as part of the vignette on how to obtain ligand-receptor reference data from CellPhoneDB from the data's original host location on GitHub. Even though this code is not evaluated, must it be removed from our vignette to meet Bioconductor standards?

Among our development group for the domino package, we deliberated between re-hosting all data used in our vignettes on Zenodo or linking to original sources of the reference annotation of ligand-receptor pairs and openly available scRNAseq data in order to credit those who have made those resources public. We have proceeded with the later option, but can adapt to only using data hosted on Zenodo if need be.

**Comment by jmitchell81** [comment]
@lshep Thank you for reviewing our submission. We have updated the package's name to dominoSignal and addressed the tasks you had set out in your prior review. The current HEAD branch at https://github.com/FertigLab/dominoSignal contains the current state of the package we would like to submit.

With the change of the package's name, can we continue the issue thread of this submission, or should I begin a new submission under the new name?

**Comment by lshep** [comment]
Please open a new issue. It will be easier to ensure all of the configuration is set up correctly.  Thank you.
